### ISSUE TITLE
Specifying a type conversion when reading raster data does not work

### ISSUE THREAD

**Issue by mmomtchev** [mmomtchev/node-gdal-async#92]
When reading raster data it should be possible to request a type conversion even with automatic buffer allocation:

```js
const data = band.pixels.read(1, 1, 20, 30, undefined, { data_type: gdal.GDT_Float64 })
```

This option is currently broken.

**Comment by mmomtchev** [comment]
The options was previously implemented as `type` but documented as `data_type`. For compatibility, both will work from now on.

**Comment by mmomtchev** [comment]
Fixed by #91
